src/RegisterDevicePage.tsx

```diff
@@ -121,44 +152,100 @@ const IntrospectionControlRow = ({
     }));
   };
 
+  const fetchInterfacesInfo = useCallback(async () => {
+    const interfacesData = await astarte.client.getInterfaces();
+    const realmManagementVersion = await astarte.client.getRealmManagementVersion();
+    const realmManagementVersionForCompare = '1.3.0';
+
+    if (semver.gte(realmManagementVersion, realmManagementVersionForCompare)) {
+      const fetchedInterfaces: FetchedInterface[] = interfacesData.map((interfaceDetail: any) => ({
+        name: interfaceDetail.interface_name,
+        major: interfaceDetail.version_major,
+        minor: interfaceDetail.version_minor,
+      }));
+      setSelectedInterfaceData(fetchedInterfaces);
+      setGreaterRealmManagementVersion(true);
+      return fetchedInterfaces;
+    } else {
+      setGreaterRealmManagementVersion(false);
+    }
+  }, [astarte.client]);
+
+  useEffect(() => {
+    const loadOptions = async () => {
+      const interfaces = await fetchInterfacesInfo();
+      if (interfaces) {
+        const interfaceOptions = interfaces.map((iface) => ({
+          value: iface.name,
+          label: iface.name,
```

Comment:
If you only display the label to the user, interface with different majors will be displayed with the same label, even if they are totally different interfaces.